torchvision/transforms/functional.py

```diff
@@ -410,10 +438,13 @@ def hflip(img):
     Returns:
         PIL Image:  Horizontall flipped image.
     """
-    if not _is_pil_image(img):
+    if not (_is_pil_image(img) or isinstance(img, torch.Tensor)):
         raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
 
-    return img.transpose(Image.FLIP_LEFT_RIGHT)
+    if _is_pil_image(img):
```

Comment:
it's much cleaner to write explicit `if`/`else`, rather than have `if` shortcut to a `return`